### PR TITLE
Planner: add action framework

### PR DESCRIPTION
**Pull request by ecourtois**
The new actions framework allows to choose the camp and select a strategy.
A strategy is composed of a set of actions containing a set of poses.
Actions and poses have callback functions to execute commands before and
after each action and each pose.
These callbacks can, for example, send messages to the robots, or generate
new actions and new poses.

### CODE REVIEW

**Review by gdoffe — COMMENTED** [review]
in commit 61123df303563b9f048a9e447df5325050564c4a message:
this will be fixed **in** a dedicated PR.

File: cogip/tools/planner/camp.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    Class representing the camp selected before the game starts.
+    """
+    class Colors(IntEnum):
+        blue = 0
+        green = 1
+
+    def __init__(self, color: Colors = Colors.blue):
+        self.color = color
+
+    def adapt_y(self, dist: float) -> float:
```
Comment (gdoffe) [inline cogip/tools/planner/camp.py:17]: Dont remember, are you sure it is Y and not X ?

File: cogip/tools/planner/planner.py [hunk 2]
```diff
@@ -94,37 +100,105 @@ def add_robot(self, robot_id: int) -> None:
...
+        Reset planner, context, robot and actions.
+        """
+        self._game_context.reset()
+        robot_ids = list(self._robots.keys())
+        self._actions = actions.action_classes.get(self._game_context.strategy, actions.Actions)()
+
+        for robot_id in robot_ids:
+            self.del_robot(robot_id)
+        for robot_id in robot_ids:
+            self.add_robot(robot_id)
```
Comment (gdoffe) [inline cogip/tools/planner/planner.py:149]: You delete robots that you just re-add after.
not sure to understand, it deserves a comment I think.

**Review by gdoffe — APPROVED** [review]
Simple & efficient ! Good job !

Just minor changes requested, I already approve.

File: cogip/tools/planner/camp.py (hunk 1 above)
Comment (ecourtois) [reply cogip/tools/planner/camp.py:17]: Yes sûre 😉

File: cogip/tools/planner/planner.py (hunk 2 above)
Comment (ecourtois) [reply cogip/tools/planner/planner.py:149]: It is a simple solution to reset robots already registered.

**Pull request merged**